### PR TITLE
Parse RCS usage mails directly

### PR DESCRIPTION
**Pull request by mbarton**
The last days usage is emailed to the Grid as a CSV attachment (via SES). This was previously parsed using a lambda with the results placed as JSON in the config bucket and then periodically read by the Grid itself.

This PR removes the lambda and moves the parsing logic into the Grid itself. The `UsageStore` periodically updates by finding the latest object in a new bucket (where the mails will be received directly) and parsing the CSV directly from the mail.

Both the `UsageStore` and `QuotaStore` are now non-optional in the code-base and always constructed, even in DEV. An example mail has been placed in both the DEV bucket for testing purposes. The TEST and PROD buckets will receive new mails, only retaining them for 7 days to minimise the impact of listing the objects.

### CODE REVIEW

File: docker/configs/generators/generators/resources/templates/dot-properties/media-api.properties.template [hunk 1]
```diff
@@ -7,6 +7,8 @@ auth.keystore.bucket={{KeyBucket}}
 sns.topic.arn={{SnsTopicArn}}
 cors.allowed.origins={{cors}}
 s3.config.bucket={{ConfigBucket}}
+s3.usagemail.bucket={{UsageMailBucket}}
 persistence.identifier=picdarUrn
 mixpanel.token={{mixpanel_token}}
 es.index.aliases.read=readAlias
+quota.store.key=rcs-quota.json
```
Comment (akash1810) [inline docker/configs/generators/generators/resources/templates/dot-properties/media-api.properties.template]: not sure we still need this property...

File: media-api/app/lib/UsageStore.scala [hunk 2]
```diff
@@ -59,29 +59,67 @@ object StoreAccess {
...
+  def csvParser(list: List[String]): List[SupplierUsageSummary] = {
+    def stripQuotes(s: String): String = s.stripSuffix("\"").stripPrefix("\"")
+
+    list.headOption match {
+      case Some(head) if head.split(',').toList.length == 2 =>
+        list.tail.map { line =>
+          val lineList = line.split(',')
+          val (supplier, count) = (
+            stripQuotes(lineList.head),
+            stripQuotes(lineList.tail.head)
```
Comment (akash1810) [inline media-api/app/lib/UsageStore.scala]: we know its a two column csv, `.tail.head` could be `.lift(2)`

Comment (akash1810) [inline media-api/app/lib/UsageStore.scala]: we could also check the values of the header row, to further ensure its the right csv we're looking for

File: media-api/test/lib/UsageStoreTest.scala [hunk 3]
```diff
@@ -0,0 +1,21 @@
...
+import org.scalatest.{FunSpec, Matchers}
+
+class UsageStoreTest extends FunSpec with Matchers {
+  describe("Usage Store") {
+    it("should parse RCS usage emails") {
+      val stream = getClass.getResourceAsStream("/example.mail")
+
+      val lines = UsageStore.extractEmail(stream)
+
+      lines.head should be ("\"Cpro Name\",\"Id\"")
```
Comment (akash1810) [inline media-api/test/lib/UsageStoreTest.scala]: `Cpro Name`?

File: docker/configs/generators/generators/resources/templates/dot-properties/media-api.properties.template (hunk 1 above)
Comment (mbarton) [inline docker/configs/generators/generators/resources/templates/dot-properties/media-api.properties.template]: vs hard-coding the name?

Comment (akash1810) [inline docker/configs/generators/generators/resources/templates/dot-properties/media-api.properties.template]: vs not having it at all. From the looks of it, this prop was used to identify what S3 key to read; if were reading the latest S3 key now we should be able to get rid of this prop. (Though I could be reading the code incorrectly... 😉 )

Comment (mbarton) [inline docker/configs/generators/generators/resources/templates/dot-properties/media-api.properties.template]: We're reading the latest for the usage store. This is for the quota store

Comment (akash1810) [inline docker/configs/generators/generators/resources/templates/dot-properties/media-api.properties.template]: d'oh! my bad

File: media-api/test/lib/UsageStoreTest.scala (hunk 3 above)
Comment (mbarton) [inline media-api/test/lib/UsageStoreTest.scala]: That's what's in the mails!

**Pull request merged**